packages/platform/atoms/timezone/index.tsx

```diff
@@ -2,22 +2,22 @@ import { TimezoneSelectComponent as TimezoneSelect } from "@calcom/features/comp
 import type { TimezoneSelectProps } from "@calcom/features/components/timezone-select";
 
 import useGetCityTimezones from "../hooks/useGetCityTimezones";
+import { filterAndFormatTimezones } from "../src/lib/filterAndFormatTimezones";
 
-export function Timezone(props: TimezoneSelectProps) {
-  const { isLoading, data } = useGetCityTimezones();
+export function Timezone(props: TimezoneSelectProps & { timeZonesFromProps?: string[] }) {
+  const { isLoading: isLoadingCityTimezones, data: cityTimezones } = useGetCityTimezones();
+  let timeZonesFiltered = undefined;
```

Comment:
I think this should be `useMemo` variable that depends on `isLoadingCityTimezones` and then it returns formatted timezones.